lib/signs_ui/alerts/state.ex

```diff
@@ -4,79 +4,83 @@ defmodule SignsUi.Alerts.State do
   them with signs to auto-expire.
   """
 
-  use GenServer
+  use GenStage
+
   require Logger
+  alias ServerSentEventStage.Event
   alias SignsUi.Alerts.Alert
+  alias SignsUi.Alerts.Display
+  alias SignsUi.Alerts.Events
 
-  defstruct alerts: %{}
-
-  @type route_id :: String.t()
-  @type alert_map :: %{Alert.id() => Alert.t()}
-  @type route_alerts_map :: %{route_id() => alert_map()}
-
-  @type state :: %__MODULE__{
-          alerts: route_alerts_map()
-        }
+  @type t :: %{Alert.id() => Alert.multi_route()}
 
+  @spec start_link(keyword) :: :ignore | {:error, any} | {:ok, pid}
   def start_link(opts \\ []) do
     {start_opts, init_opts} = Keyword.split(opts, [:name])
-    start_opts = Keyword.put_new(start_opts, :name, __MODULE__)
-    GenServer.start_link(__MODULE__, init_opts, start_opts)
+    GenStage.start_link(__MODULE__, init_opts, start_opts)
   end
 
+  @impl GenStage
   def init(opts) do
-    interval_ms = Keyword.get(opts, :interval_ms, 15_000)
-    :timer.send_interval(interval_ms, :twiddle_state)
-    {:ok, %__MODULE__{}}
+    {:consumer, %{}, opts}
   end
 
-  @spec active_alert_ids(GenServer.server()) :: MapSet.t(Alert.id())
+  @impl GenStage
+  def handle_info(msg, state) do
```

Comment:
Actually, I think you can just delete this. Basically, we need to provide a fallback `handle_info` to handle unexpected messages, if we define our own `handle_info` for certain messages, which I had done with `:twiddle_state`. However, now that `:twiddle_state` is gone, we can remove this fallback. If we don't define *any* `handle_info` in the module, the `use GenStage` macro provides a reasonable default like this for us.